### ISSUE TITLE
Unsatisfied dependency for type `javax.sql.DataSource` when using JDBC Postgres datasource

### ISSUE THREAD

**Issue by Fryuni** [quarkusio/quarkus#43815]
### Describe the bug

Consuming a DataSource using the `javax.sql.DataSource` class doesn't work, contrary to the documentation:

> In the above example, the type is `AgroalDataSource`, a `javax.sql.DataSource` subtype. Because of this, you can also use `javax.sql.DataSource` as the injected type.

### Expected behavior

Configured DataSource should be injected on the property with type `javax.sql.DataSource`

### Actual behavior

```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.13.0:build (default) on project internal-project: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type javax.sql.DataSource and qualifiers [@Default]
[ERROR] 	- injection target: parameter 'defaultDataSource' of com.croct.project.ProjectClass constructor
[ERROR] 	- declared on CLASS bean [types=[com.croct.project.ProjectClass java.lang.Object], qualifiers=[@Default, @Any], target=com.croct.project.ProjectClass]
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1551)
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:338)
[ERROR] 	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:167)
[ERROR] 	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:490)
[ERROR] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1575)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
[ERROR] Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type javax.sql.DataSource and qualifiers [@Default]
[ERROR] 	- injection target: parameter 'defaultDataSource' of com.croct.project.ProjectClass constructor
[ERROR] 	- declared on CLASS bean [types=[com.croct.project.ProjectClass, java.lang.Object], qualifiers=[@Default, @Any], target=com.croct.project.ProjectClass]
[ERROR] 	at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:545)
[ERROR] 	at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:667)
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:323)
[ERROR] 	... 12 more
```

### How to Reproduce?

Reproducer:
```java
@ApplicationScoped
public final class ProjectClass {
    private final ApplicationConfig configuration;

    private final DataSource dataSource;

    @Inject
    public ProjectClass(
        final ApplicationConfig configuration,
        final DataSource defaultDataSource
    ) {
        this.configuration = configuration;
        this.dataSource = defaultDataSource;
    }
}
```

Then run `mvnw -DskipTests clean package`

### Output of `uname -a` or `ver`

Darwin Fry-MacBook-x86.local 23.6.0 Darwin Kernel Version 23.6.0: Mon Jul 29 21:13:00 PDT 2024; root:xnu-10063.141.2~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk 23 2024-09-17 OpenJDK Runtime Environment Homebrew (build 23) OpenJDK 64-Bit Server VM Homebrew (build 23, mixed mode, sharing)

### Quarkus version or git rev

3.13.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.8 (36645f6c9b5079805ea5009217e36f2cffd34256) Maven home: /Users/lotus/.m2/wrapper/dists/apache-maven-3.9.8-bin/337e6d14/apache-maven-3.9.8 Java version: 23, vendor: Homebrew, runtime: /usr/local/Cellar/openjdk/23/libexec/openjdk.jdk/Contents/Home Default locale: en_BR, platform encoding: UTF-8 OS name: "mac os x", version: "14.6.1", arch: "x86_64", family: "mac"

### Additional information

```
[INFO] --- quarkus:3.13.0:info (default-cli) @ user-updater ---
[WARNING] quarkus:info goal is experimental, its options and output may change in future versions
[INFO] Quarkus platform BOMs:
[INFO]         io.quarkus.platform:quarkus-bom:pom:3.13.0 ✔
[INFO]
[INFO] Extensions from io.quarkus.platform:quarkus-bom:
[INFO]         io.quarkus:quarkus-confluent-registry-avro ✔
[INFO]         io.quarkus:quarkus-kafka-streams ✔
[INFO]         io.quarkus:quarkus-smallrye-health ✔
[INFO]         io.quarkus:quarkus-jdbc-postgresql ✔
[INFO]         io.quarkus:quarkus-arc ✔
[INFO]         io.quarkus:quarkus-micrometer ✔
[INFO]         io.quarkus:quarkus-micrometer-registry-prometheus ✔
```

**Comment by yrodiere** [comment]
Thanks for reporting.

FWIW the bean _is_ registered with the `javax.sql.DataSource` type: https://github.com/quarkusio/quarkus/blame/54a9feff06744b728cbc65c4e2b680b234e6473d/extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java#L284

In your case, I think the problem is simply that you're not using the Agroal extension, which is the one registering `DataSource` beans. The root cause is similar to that of #43428, and the fix would be the same: #43685 , coming in Quarkus 3.16.

As a workaround if you want to use Quarkus 3.15 or earlier, just add a dependency to `io.quarkus:quarkus-agroal` in your application.

Closing as a duplicate (sort of) of #43428.

**Comment by Fryuni** [comment]
> In your case, I think the problem is simply that you're not using the Agroal extension, which is the one registering DataSource beans.

So should the docs be updated? It states this:

> Using a built-in JDBC driver extension automatically includes the Agroal extension, which is the JDBC connection pool implementation applicable for custom and built-in JDBC drivers.

Following the docs leads to this error

**Comment by yrodiere** [comment]
> > In your case, I think the problem is simply that you're not using the Agroal extension, which is the one registering DataSource beans.
> 
> So should the docs be updated? It states this:
> 
> > Using a built-in JDBC driver extension automatically includes the Agroal extension, which is the JDBC connection pool implementation applicable for custom and built-in JDBC drivers.
> 
> Following the docs leads to this error

Like I said it's fixed by #43685 : after this patch the documentation is actually correct.

I suppose we could alter the documentation for older versions that don't have this patch, but TBH at this point I'd rather backport #43685. WDYT @gsmet , should we add the dependency from JDBC drivers to Agroal in Quarkus 3.15, or even older versions?
